plugins/inputs/prometheus/prometheus.go

```diff
@@ -63,6 +79,28 @@ func (p *Prometheus) Description() string {
 
 var ErrProtocolError = errors.New("prometheus protocol error")
 
+func (p *Prometheus) SplitHostAndPort(host string) (string, string) {
```

Comment:
You can use URL.Hostname() and URL.Port() for this.

Reply:
Done!